sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -1165,14 +1175,29 @@ public ResourceId apply(String input) {
       if (tempDirectory == null) {
         tempDirectory = getFilenamePrefix();
       }
-      WriteFiles<UserT, DestinationT, OutputT> write =
-          WriteFiles.to(
-              new AvroSink<>(tempDirectory, resolveDynamicDestinations(), getGenericRecords()));
+      final DynamicAvroDestinations<UserT, DestinationT, OutputT> dynamicDestinations =
+          resolveDynamicDestinations();
+
+      FileIO.Write<DestinationT, UserT> write =
+          FileIO.<DestinationT, UserT>writeDynamic()
+              .by(new DestinationFnViaDynamicDestinations<>(dynamicDestinations))
+              .via(
```

Comment:
<!--new_thread; commit:28d55b3a9ab8a0cd84e7834b34c2c150edf871d7; resolved:0-->
Will this cache sinks per destination, or constantly recreate sinks? If the latter., this might be a performance regression.